src/GovUk.Education.ExploreEducationStatistics.Publisher/Functions/PublishScheduledReleasesFunction.cs

```diff
@@ -0,0 +1,77 @@
+#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Publisher.Model;
+using GovUk.Education.ExploreEducationStatistics.Publisher.Services.Interfaces;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using static GovUk.Education.ExploreEducationStatistics.Publisher.Model.ReleasePublishingStatusPublishingStage;
+
+namespace GovUk.Education.ExploreEducationStatistics.Publisher.Functions
+{
+    // ReSharper disable once UnusedType.Global
+    public class PublishScheduledReleasesFunction
+    {
+        private readonly IReleasePublishingStatusService _releasePublishingStatusService;
+        private readonly IPublishingService _publishingService;
+        private readonly IPublishingCompletionService _publishingCompletionService;
+
+        public PublishScheduledReleasesFunction(
+            IReleasePublishingStatusService releasePublishingStatusService,
+            IPublishingService publishingService,
+            IPublishingCompletionService publishingCompletionService)
+        {
+            _releasePublishingStatusService = releasePublishingStatusService;
+            _publishingService = publishingService;
+            _publishingCompletionService = publishingCompletionService;
+        }
+
+        /// <summary>
+        /// Azure function which publishes the content for a Release at a scheduled time by moving it from a staging directory.
+        /// </summary>
+        /// <remarks>
+        /// Sets the published time on the Release which means it's considered as 'Live'.
+        /// </remarks>
+        /// <param name="timer"></param>
+        /// <param name="executionContext"></param>
+        /// <param name="logger"></param>
+        /// <returns></returns>
+        [FunctionName("PublishStagedReleaseContent")]
+        // ReSharper disable once UnusedMember.Global
+        public async Task PublishScheduledReleases([TimerTrigger("%PublishReleaseContentCronSchedule%")]
+            TimerInfo timer,
+            ExecutionContext executionContext,
+            ILogger logger)
+        {
+            logger.LogInformation("{FunctionName} triggered at: {DateTime}",
+                executionContext.FunctionName,
+                DateTime.UtcNow);
+
+            var scheduled = (await QueryScheduledReleases()).ToArray();
+            if (scheduled.Any())
+            {
+                // Move all cached releases in the staging directory of the public content container to the root
+                await _publishingService.PublishStagedReleaseContent();
```

Comment:
Aww maaan, I think that I would love to get rid of it altogether, but I don't like losing that early warning system!

To be honest, I like the idea of being able to just generate this content immediately in such a way that it could be in its public "live" storage location immediately, but versioned somehow so that if it's a brand new Release, we only serve it up when its target Published date has passed, or if it's an amendment, we switch over to serving it rather than its previous version when its target Published date has passed (and then somehow clean up the old Release's "live" files afterwards).

That way, we still have the early warning system, but we don't have the need to move staged data about.  Instead, it would almost be like updating a "storage symlink" that points to a Release's "current" content location in storage.  A quick Google shows that there's probably no mechanism for anything like symlinks in Azure Storage at the moment though.

I think that either way, for now we should leeave any changes like this out of this particular PR, but it might be worth chewing over for the cleanup PR...